src/Hadrian/Builder/Type.hs

```diff
@@ -0,0 +1,26 @@
+module Hadrian.Builder.Type where
```

Comment:
I'm a bit unhappy about this module: do we really need to separate the definition of builder modes and builder arguments? This looks very awkward.